rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/RepositoryGraphImpl.java

```diff
@@ -0,0 +1,172 @@
+	public void remove(Triple tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(statement, contextMask);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public void clear() {
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.clear(contextMask);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public long size() {
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			if (! includeInferred && contextMask.length == 0) { 
+				return conn.size();
+			} else {
+				return stream().count();
+			}
+		}
+	}
+
+	
+	@Override
+	public void add(BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.add(subj, pred, obj, contextMask);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public boolean contains(BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.hasStatement(subj, pred, obj, includeInferred, contextMask);
+		}
+	}
+
+	@Override
+	public void remove(BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(subj, pred, obj, contextMask);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public Stream<RDF4JTriple> stream() {
+		return stream(null, null, null);
+	}
+	
+	@Override
+	public Stream<RDF4JTriple> stream(BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		RepositoryConnection conn = getRepositoryConnection();
+		// FIXME: Is it OK that we don't close the connection?
+		RepositoryResult<Statement> statements = conn.getStatements(subj, pred, obj, includeInferred, contextMask);
+		return Iterations.stream(statements).map(this::asTripleLike);
+	}
+	
+	@Override
+	protected RDF4JTriple asTripleLike(Statement statement) {
+		return rdf4jTermFactory.asTriple(statement);
+	}
+
+	@Override
+	public Set<RDF4JBlankNodeOrIRI<Resource>> getContextMask() {		
+		Set<RDF4JBlankNodeOrIRI<Resource>> mask = new HashSet<>();
```

Comment:
Not sure how often this method will be used, but this set should not become very large in any normal situations, compared to the rest of the data, so this could be prederived in the constructor and stored in an instance variable next to contextMask.